src/Service/PaymentFunctions.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+namespace Drupal\commerce_payment_dibs\Service;
+
+use Drupal\commerce_order\Entity\Order;
+use Drupal\commerce_payment\Entity\Payment;
+use Drupal\commerce_payment\Entity\PaymentGateway;
+use Drupal\commerce_payment_dibs\DibsTransactionService;
+use GuzzleHttp\Client;
+
+/**
+ * Class PaymentFunctions.
+ *
+ * @package Drupal\commerce_payment_dibs\Service
+ */
+class PaymentFunctions {
+
+  /**
+   * Cancel url.
+   */
+  const CANCEL_URL = 'https://payment.architrade.com/cgi-adm/cancel.cgi';
+
+  /**
+   * @var \Drupal\commerce_payment_dibs\DibsTransactionService
+   */
+  protected $transactionService;
+
+  /**
+   * @var \GuzzleHttp\Client
+   */
+  protected $client;
+
+  /**
+   * @var array
+   */
+  protected $gatewayConfiguration;
+
+  /**
+   * @var array
+   */
+  protected $lastResponse;
+
+  /**
+   * PaymentFunctions constructor.
+   *
+   * @param \Drupal\commerce_payment_dibs\DibsTransactionService $transactionService
+   *   Transaction service.
+   * @param \GuzzleHttp\Client $client
+   *   Client.
+   */
+  public function __construct(DibsTransactionService $transactionService, Client $client) {
+    $this->transactionService = $transactionService;
+    $this->client = $client;
+
+    $payment_gateway = PaymentGateway::load('dibs');
+    $payment_gateway_plugin = $payment_gateway->getPlugin();
```

Comment:
We might want to generalise this since we cannot guarantee that the gateway will be called dibs. Just an improvement thought.